pettingzoo/butterfly/pistonball/pistonball.py

```diff
@@ -593,19 +596,27 @@ def get_local_reward(self, prev_position, curr_position):
         local_reward = 0.5 * (prev_position - curr_position)
         return local_reward
 
-    def render(self, mode="human"):
-        if mode == "human" and not self.renderOn:
+    def render(self):
+        if self.render_mode is None:
+            gym.logger.WARN(
+                "You are calling render method without specifying any render mode."
+            )
+            return
+
+        if self.render_mode == "human" and not self.renderOn:
```

Comment:
I see these lines a lot and I'm wondering if this should be in the main `AECEnv` and `ParallelEnv` class itself. Thoughts?

Reply:
Yes, it would be nice to avoid repetitions
However, if I put it on the superclass with every subclass calling super().render(), then `return` will not be effective to exit from render (it will exit only from the super().render())
The second small reason is that `AECEnv` and `ParallelEnv` are supposed to be abstract classes and `render` an abstract method. For example, if while writing a custom environment you forgot to implement `render`, PyCharm will highlight it, and calling render will raise an error (instead of doing nothing).

Reply:
This is a fair argument, so we are only asserting that _our_ environments have a `render_mode`, and the user-created envs don't have this requirement? If this is the case, this can be resolved. Paging @pseudo-rnd-thoughts for clarity.

Reply:
I guess it depends on how backward compatible you want to be. 
I may be missing something, but will this check only be run in PZ envs currently

Reply:
@jjshoots The good practice for users is to always define it, because Wrapper, for example, can use this attribute. 
I can assign `render_mode=None` in the superclasses `AECEnv` and `ParallelEnv`, the environments should override it. 

Btw, I noticed this is not done for metadata, for example (that also should be defined). Do you know why? See https://github.com/openai/gym/blob/e055ee7b8e7a95d5cb41c2cb5bfd3603c67b516c/gym/core.py#L62 for example

Reply:
@pseudo-rnd-thoughts Yes, right now it's only for PZ envs.
@younik Personally, I'm inclined to not have it defined in the superclasses if this is something that we're not enforcing in the superclasses.

Reply:
I think this is fine, this might be solved in PZ2 but we can make a decision then about it

Reply:
Ok, marking this as resolved then.